queries/javascript.core.scm

```diff
@@ -415,6 +404,262 @@
+(new_expression
+  (arguments) @functionCallee.end.startOf
+) @functionCallee.start.startOf @_.domain
+
+;;!! class Foo {}
+;;!  ^^^^^^^^^^^^
+(
+  [
+    (class_declaration)
+    (class)
+  ] @class
+  (#not-parent-type? @class export_statement)
+)
+
+;;!! export class Foo {}
+;;!  ^^^^^^^^^^^^^^^^^^^
+(export_statement
+  [
+    (class_declaration)
+    (class)
+  ]
+) @class
+
+;;!! class Foo {}
+;;!        ^^^
+;;!  ------------
+(class_declaration
+  name: (_) @className
+) @_.domain
+(class
+  name: (_) @className
+) @_.domain
+
+;;!! true ? 0 : 1;
+;;!  ^^^^
+;;!         ^   ^
+;;! --------------
+(ternary_expression
+  condition: (_) @condition
+  consequence: (_) @branch
+) @condition.domain
+(ternary_expression
+  alternative: (_) @branch
+)
+
+;;!! for (let i = 0; i < 2; ++i) {}
+;;!                  ^^^^^
+;;!  ------------------------------
+(for_statement
+  condition: (_) @condition
+  (#child-range! @condition 0 -1 false true)
+) @_.domain
+
+;;!! while (true) {}
+;;!         ^^^^
+(while_statement
+  condition: (_) @condition
+  (#child-range! @condition 0 -1 true true)
+) @_.domain
+
+;;!! do {} while (true);
+;;!               ^^^^
+(do_statement
+  condition: (_) @condition
+  (#child-range! @condition 0 -1 true true)
+) @_.domain
+
+;;!! case 0: {}
+;;!  ^^^^^^^^^^
+;;!       ^
+(switch_case
+  value: (_) @condition
+) @branch @condition.domain
+
+;;!! switch () {}
+;;!  ^^^^^^^^^^^^
+(switch_statement) @branch.iteration @condition.iteration
+
+;;!! if () {}
+;;!  ^^^^^^^^
+(
+  (if_statement
+    condition: (_) @condition
+    consequence: (_) @branch.end.endOf
+  ) @branch.start.startOf @condition.domain
+  (#not-parent-type? @condition.domain "else_clause")
+  (#child-range! @condition 0 -1 true true)
+)
+
+;;!! else if () {}
+;;!  ^^^^^^^^^^^^^
+(else_clause
+  (if_statement
+    condition: (_) @condition
+    consequence: (_) @branch.end.endOf @condition.domain.end.endOf
+  )
+  (#child-range! @condition 0 -1 true true)
+) @branch.start.startOf @condition.domain.start.startOf
+
+;;!! else {}
+;;!  ^^^^^^^
+(else_clause
+  (statement_block)
+) @branch
+
+;;!! if () {} else if () {} else {}
+;;!  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+(
+  (if_statement) @branch.iteration
+  (#not-parent-type? @branch.iteration "else_clause")
+)
+
+;;!! try () {}
+;;!  ^^^^^^^^^
+(try_statement
+  "try" @branch.start
+  body: (_) @branch.end
+)
+
+;;!! catch () {}
+;;!  ^^^^^^^^^^^
+(try_statement
+  handler: (_) @branch
+)
+
+;;!! finally {}
+;;!  ^^^^^^^^^^
+(try_statement
+  finalizer: (_) @branch
+)
+
+;;!! try () {} catch () {} finally {}
+;;!  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+(try_statement) @branch.iteration
+
+;;!! { value: 0 }
+;;!    ^^^^^
+;;!    xxxxxxx
+;;!! { value: 0 }
+;;!           ^
+;;!         xxx
+;;!    --------
+(pair
+  key: (_) @collectionKey @collectionKey.removal.start @value.removal.start.endOf
+  value: (_) @value @collectionKey.removal.end.startOf @value.removal.end
+) @_.domain
+
+;; Non-exported statements
```

Comment:
Technically this includes exported statements too

Reply:
true